packages/driver/src/cross-origin/communicator.ts

```diff
@@ -11,6 +11,51 @@ const CROSS_ORIGIN_PREFIX = 'cross:origin:'
 const LOG_EVENTS = [`${CROSS_ORIGIN_PREFIX}log:added`, `${CROSS_ORIGIN_PREFIX}log:changed`]
 const SNAPSHOT_EVENT_PREFIX = `${CROSS_ORIGIN_PREFIX}snapshot:`
 
+/**
+ * Shared Promise Setup, is a helper function to setup our promisified post messages.
+ * @param resolve the promise resolve function
+ * @param reject the promise reject function
+ * @param data the data to send
+ * @param event the name of the event to be promisified.
+ * @param specBridgeName the name of the spec bridge receiving the event.
+ * @param communicator the communicator that is sending the message
+ * @returns the data to send
+ */
+const sharedPromiseSetup = ({ resolve,
+  reject,
+  data,
+  event,
+  specBridgeName,
+  communicator,
+}: {
+  resolve: Function
+  reject: Function
+  data?: any
+  event: string
+  specBridgeName: string
+  communicator: EventEmitter
+}) => {
+  let timeoutId
+
+  const dataToSend = data || {}
+
+  dataToSend.specBridgePromiseId = Date.now()
```

Comment:
Is this unique enough for this use case? I think so??? I didn't want to pull in a uuid library just for this.

Reply:
maybe an eventName + Date.now() composite?

Reply:
That is what i use this to listen to, i suppose it could just send that as the event to reflect to, that would be easier.

Reply:
Ah I see that below now 😆 . I feel like that _should_ be enough. we have done similar unique IDs in the past